src/api/app/models/concerns/staging_project.rb

```diff
@@ -87,19 +85,30 @@ def broken_packages
     @broken_packages
   end
 
-  def missing_reviews
-    return @missing_reviews if @missing_reviews
-
-    @missing_reviews = []
+  def missing_reviews_for_classified_requests(request, reviews)
+    @missing_reviews_of_st_project ||= []
 
-    Review.includes(:bs_request).where(bs_request_id: staged_requests.select(:id)).where.not(state: :accepted).find_each do |review|
-      # We skip reviews for the staging project since these reviews are used
-      # by the openSUSE release tools _after_ the overall_state switched to
-      # 'accepted'.
+    reviews.each_with_object([]) do |review, collected|
       next if review.by_project == name
```

Comment:
@vpereira filtering out the reviews where by_project is the staging project itself, as part of the database query turned out to be a bit difficult if you want to prevent the n+1 queries. With the ActiveRecord Preloader it was very convenient, but like you discovered not stable. Every other attempt so far ended up in either unreadable code or worse performance. I think its not worth to spend more time on that. Everything else besides that part is now filtered in the query. In the end it wouldn't bring a huge performance boost anyway.